libflux/go/libflux/analyze.go

```diff
@@ -0,0 +1,35 @@
+// +build libflux
+
+package libflux
+
+// #cgo CFLAGS: -I.
+// #cgo LDFLAGS: -L. -llibstd
```

Comment:
My understanding was that the `libstd` crate would be internal and it would only be used for generating a file in the `flux` crate and we would use conditional compilation to bootstrap the `flux` crate.

Reply:
In the below code added to reference the prelude, I am not seeing a reference to `libstd`. Are you sure that this is needed?

Reply:
Hmm, that doesn't seem to be how it works though.  We generate `.data` files that contain serialized types for the standard library and prelude, but they are consumed by `libstd` which exposes public `prelude` and `stdlib` functions.  The `libstd` crate depends on  the `flux` crate, so I don't see how to type infer Flux and include the standard library unless Go talks to the `libstd` crate.

Maybe @jlapacik can shed more light on how this is intended to work.

I'm not sure what you mean by "the below code added to reference the prelude"... maybe I'm misunderstanding your question.

Reply:
The original idea @jlapacik and I worked out is that the `flux` crate would conditionally compile. The `libstd` crate would use a `nostdlib` or `core-only` feature flag when referencing flux and would generate the data files which would only be referenced when the standard library feature flag was activated. The default would end up being the standard library as enabled. This way, we could have `libstd` reference the `flux` crate while avoiding multiple crates. This is why I figured the linker flag for this should still be to `libflux` because `libflux` is supposed to contain everything. I am not sure if this got implemented in practice so I apologize if I have this incorrect.